### ISSUE TITLE
Can't sync project by Control Plane Execution Environment

### ISSUE THREAD

**Issue by lutskevich** [ansible/awx#12117]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.

### Summary

https://quay.io/repository/ansible/awx-ee?tab=tags&tag=latest

awx-ee:latest released 7 hours ago, and now Control Plane Execution Environment (quay.io/ansible/awx-ee:latest) can't sync project

### AWX version

21.0.0

### Select the relevant components

- [ ] UI
- [ ] API
- [ ] Docs

### Installation method

minikube

### Modifications

no

### Steps to reproduce

Sync project

### Expected results

Project synced

### Actual results

{
  "changed": false,
  "stdout": "the full traceback was:\n\nTraceback (most recent call last):\n  File \"/usr/local/bin/ansible-galaxy\", line 97, in <module>\n    mycli = getattr(__import__(\"ansible.cli.%s\" % sub, fromlist=[myclass]), myclass)\n  File \"/usr/local/lib/python3.8/site-packages/ansible/cli/galaxy.py\", line 24, in <module>\n    from ansible.galaxy.api import GalaxyAPI\n  File \"/usr/local/lib/python3.8/site-packages/ansible/galaxy/api.py\", line 28, in <module>\n    from ansible.module_utils.urls import open_url, prepare_multipart\n  File \"/usr/local/lib/python3.8/site-packages/ansible/module_utils/urls.py\", line 115, in <module>\n    from urllib3.contrib.pyopenssl import PyOpenSSLContext\n  File \"/usr/lib/python3.8/site-packages/urllib3/contrib/pyopenssl.py\", line 46, in <module>\n    import OpenSSL.SSL\n  File \"/usr/local/lib/python3.8/site-packages/OpenSSL/__init__.py\", line 8, in <module>\n    from OpenSSL import crypto, SSL\n  File \"/usr/local/lib/python3.8/site-packages/OpenSSL/crypto.py\", line 1517, in <module>\n    class X509StoreFlags(object):\n  File \"/usr/local/lib/python3.8/site-packages/OpenSSL/crypto.py\", line 1537, in X509StoreFlags\n    CB_ISSUER_CHECK = _lib.X509_V_FLAG_CB_ISSUER_CHECK\nAttributeError: module 'lib' has no attribute 'X509_V_FLAG_CB_ISSUER_CHECK'",
  "stderr": "ERROR! Unexpected Exception, this is probably a bug: module 'lib' has no attribute 'X509_V_FLAG_CB_ISSUER_CHECK'",
  "rc": 250,
  "cmd": [
    "ansible-galaxy",
    "role",
    "install",
    "-r",
    "/var/lib/awx/projects/_8__all_project/roles/requirements.yml",
    "--roles-path",
    "/var/lib/awx/projects/.__awx_cache/_8__all_project/stage/requirements_roles"
  ],
  "start": "2022-04-27 07:09:18.665963",
  "end": "2022-04-27 07:09:19.132270",
  "delta": "0:00:00.466307",
  "msg": "non-zero return code",
  "invocation": {
    "module_args": {
      "chdir": "/var/lib/awx/projects/_8__all_project",
      "_raw_params": "ansible-galaxy role install -r /var/lib/awx/projects/_8__all_project/roles/requirements.yml --roles-path /var/lib/awx/projects/.__awx_cache/_8__all_project/stage/requirements_roles \n",
      "_uses_shell": false,
      "warn": false,
      "stdin_add_newline": true,
      "strip_empty_ends": true,
      "argv": null,
      "executable": null,
      "creates": null,
      "removes": null,
      "stdin": null
    }
  },
  "stdout_lines": [
    "the full traceback was:",
    "",
    "Traceback (most recent call last):",
    "  File \"/usr/local/bin/ansible-galaxy\", line 97, in <module>",
    "    mycli = getattr(__import__(\"ansible.cli.%s\" % sub, fromlist=[myclass]), myclass)",
    "  File \"/usr/local/lib/python3.8/site-packages/ansible/cli/galaxy.py\", line 24, in <module>",
    "    from ansible.galaxy.api import GalaxyAPI",
    "  File \"/usr/local/lib/python3.8/site-packages/ansible/galaxy/api.py\", line 28, in <module>",
    "    from ansible.module_utils.urls import open_url, prepare_multipart",
    "  File \"/usr/local/lib/python3.8/site-packages/ansible/module_utils/urls.py\", line 115, in <module>",
    "    from urllib3.contrib.pyopenssl import PyOpenSSLContext",
    "  File \"/usr/lib/python3.8/site-packages/urllib3/contrib/pyopenssl.py\", line 46, in <module>",
    "    import OpenSSL.SSL",
    "  File \"/usr/local/lib/python3.8/site-packages/OpenSSL/__init__.py\", line 8, in <module>",
    "    from OpenSSL import crypto, SSL",
    "  File \"/usr/local/lib/python3.8/site-packages/OpenSSL/crypto.py\", line 1517, in <module>",
    "    class X509StoreFlags(object):",
    "  File \"/usr/local/lib/python3.8/site-packages/OpenSSL/crypto.py\", line 1537, in X509StoreFlags",
    "    CB_ISSUER_CHECK = _lib.X509_V_FLAG_CB_ISSUER_CHECK",
    "AttributeError: module 'lib' has no attribute 'X509_V_FLAG_CB_ISSUER_CHECK'"
  ],
  "stderr_lines": [
    "ERROR! Unexpected Exception, this is probably a bug: module 'lib' has no attribute 'X509_V_FLAG_CB_ISSUER_CHECK'"
  ],
  "_ansible_no_log": false,
  "item": "/var/lib/awx/projects/_8__all_project/roles/requirements.yml",
  "ansible_loop_var": "item",
  "_ansible_item_label": "/var/lib/awx/projects/_8__all_project/roles/requirements.yml"
}

**Comment by JonathanG3** [comment]
Hopping in as I'm also getting an issue with the default EE, but it's a bit different:
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/ansible_runner/__main__.py", line 901, in main
    res = run(**run_options)
  File "/usr/local/lib/python3.8/site-packages/ansible_runner/interface.py", line 216, in run
    r.run()
  File "/usr/local/lib/python3.8/site-packages/ansible_runner/streaming.py", line 130, in run
    r = ansible_runner.interface.run(**self.kwargs)
  File "/usr/local/lib/python3.8/site-packages/ansible_runner/interface.py", line 215, in run
    r = init_runner(**kwargs)
  File "/usr/local/lib/python3.8/site-packages/ansible_runner/interface.py", line 123, in init_runner
    rc = RunnerConfig(**kwargs)
  File "/usr/local/lib/python3.8/site-packages/ansible_runner/config/runner.py", line 79, in __init__
    super(RunnerConfig, self).__init__(private_data_dir, **kwargs)
TypeError: __init__() got an unexpected keyword argument 'suppress_env_files'


  This can be caused if the version of ansible-runner in your execution environment is out of date.
```

** EDITITING ** with some stuff that helped me solve this issue:
1. I had to add some pip upgrades to my custom awx-ee image (that I already had) but these are the packages that mattered:
```
RUN python3 -m pip install --upgrade --user -r requirements.txt
```
```
cryptography>=35.0,<37.0.0
pyopenssl
ansible-runner
```
2. as I'm using the awx-operator I had to change the default control_plane_ee_image to use my image instead of the official one (quay.io/ansible/awx-ee:latest) so in the spec paragraph in the operator spec:
```
spec:
  ....
  control_plane_ee_image: <Private Repository>/custom-awx-ee:latest
  ...
```

I will return it to the official image once it's fixed but for now I'll use my own, it's just that my own image has a lot more and I want the project syncs to be in a clean AWX-EE image.

Hopefully that will help someone.

**Comment by ivantomica** [comment]
Perhaps related to:
- https://github.com/ansible/awx-ee/issues/112

**Comment by aporwal3** [comment]
So how can we downgrade cryptography to 36.0.2 ? Currently am on AWX 20.1.0

**Comment by lutskevich** [comment]
Dockerfile
FROM quay.io/ansible/awx-ee:latest
RUN pip3 install cryptography==36.0.2

Work fine

**Comment by aporwal3** [comment]
@lutskevich tried with customised awx-ee image (the one that you suggested above) but still am getting the same error message on AWX 20.1.0.

Attaching the logs below for reference

{
  "changed": false,
  "stdout": "the full traceback was:\n\nTraceback (most recent call last):\n  File \"/usr/local/bin/ansible-galaxy\", line 97, in <module>\n    mycli = getattr(__import__(\"ansible.cli.%s\" % sub, fromlist=[myclass]), myclass)\n  File \"/usr/local/lib/python3.8/site-packages/ansible/cli/galaxy.py\", line 24, in <module>\n    from ansible.galaxy.api import GalaxyAPI\n  File \"/usr/local/lib/python3.8/site-packages/ansible/galaxy/api.py\", line 28, in <module>\n    from ansible.module_utils.urls import open_url, prepare_multipart\n  File \"/usr/local/lib/python3.8/site-packages/ansible/module_utils/urls.py\", line 115, in <module>\n    from urllib3.contrib.pyopenssl import PyOpenSSLContext\n  File \"/usr/lib/python3.8/site-packages/urllib3/contrib/pyopenssl.py\", line 46, in <module>\n    import OpenSSL.SSL\n  File \"/usr/local/lib/python3.8/site-packages/OpenSSL/__init__.py\", line 8, in <module>\n    from OpenSSL import crypto, SSL\n  File \"/usr/local/lib/python3.8/site-packages/OpenSSL/crypto.py\", line 1517, in <module>\n    class X509StoreFlags(object):\n  File \"/usr/local/lib/python3.8/site-packages/OpenSSL/crypto.py\", line 1537, in X509StoreFlags\n    CB_ISSUER_CHECK = _lib.X509_V_FLAG_CB_ISSUER_CHECK\nAttributeError: module 'lib' has no attribute 'X509_V_FLAG_CB_ISSUER_CHECK'",
  "stderr": "ERROR! Unexpected Exception, this is probably a bug: module 'lib' has no attribute 'X509_V_FLAG_CB_ISSUER_CHECK'",
  "rc": 250,
  "cmd": [
    "ansible-galaxy",
    "collection",
    "install",
    "-r",
    "/var/lib/awx/projects/_189__project_0441_pm/collections/requirements.yml",
    "--collections-path",
    "/var/lib/awx/projects/.__awx_cache/_189__project_0441_pm/stage/requirements_collections",
    "-vvv"
  ],
  "start": "2022-04-27 12:20:56.036533",
  "end": "2022-04-27 12:20:56.515648",
  "delta": "0:00:00.479115",
  "msg": "non-zero return code",
  "invocation": {
    "module_args": {
      "chdir": "/var/lib/awx/projects/_189__project__0441_pm",
      "_raw_params": "ansible-galaxy collection install -r /var/lib/awx/projects/_189__project__0441_pm/collections/requirements.yml --collections-path /var/lib/awx/projects/.__awx_cache/_189__project__0441_pm/stage/requirements_collections  -vvv\n",
      "_uses_shell": false,
      "warn": false,
      "stdin_add_newline": true,
      "strip_empty_ends": true,
      "argv": null,
      "executable": null,
      "creates": null,
      "removes": null,
      "stdin": null
    }
  },
  "stdout_lines": [
    "the full traceback was:",
    "",
    "Traceback (most recent call last):",
    "  File \"/usr/local/bin/ansible-galaxy\", line 97, in <module>",
    "    mycli = getattr(__import__(\"ansible.cli.%s\" % sub, fromlist=[myclass]), myclass)",
    "  File \"/usr/local/lib/python3.8/site-packages/ansible/cli/galaxy.py\", line 24, in <module>",
    "    from ansible.galaxy.api import GalaxyAPI",
    "  File \"/usr/local/lib/python3.8/site-packages/ansible/galaxy/api.py\", line 28, in <module>",
    "    from ansible.module_utils.urls import open_url, prepare_multipart",
    "  File \"/usr/local/lib/python3.8/site-packages/ansible/module_utils/urls.py\", line 115, in <module>",
    "    from urllib3.contrib.pyopenssl import PyOpenSSLContext",
    "  File \"/usr/lib/python3.8/site-packages/urllib3/contrib/pyopenssl.py\", line 46, in <module>",
    "    import OpenSSL.SSL",
    "  File \"/usr/local/lib/python3.8/site-packages/OpenSSL/__init__.py\", line 8, in <module>",
    "    from OpenSSL import crypto, SSL",
    "  File \"/usr/local/lib/python3.8/site-packages/OpenSSL/crypto.py\", line 1517, in <module>",
    "    class X509StoreFlags(object):",
    "  File \"/usr/local/lib/python3.8/site-packages/OpenSSL/crypto.py\", line 1537, in X509StoreFlags",
    "    CB_ISSUER_CHECK = _lib.X509_V_FLAG_CB_ISSUER_CHECK",
    "AttributeError: module 'lib' has no attribute 'X509_V_FLAG_CB_ISSUER_CHECK'"
  ],
  "stderr_lines": [
    "ERROR! Unexpected Exception, this is probably a bug: module 'lib' has no attribute 'X509_V_FLAG_CB_ISSUER_CHECK'"
  ],
  "_ansible_no_log": false,
  "item": "/var/lib/awx/projects/_189__project__0441_pm/collections/requirements.yml",
  "ansible_loop_var": "item",
  "_ansible_item_label": "/var/lib/awx/projects/_189__project__0441_pm/collections/requirements.yml"
}

**Comment by Jonathan-Caruana** [comment]
Hi everyone,

Same problem on my side, hope this will be fixed soon :) 

I noticed that the awx-ee image on quay.io was updated 45min ago but even if i use this image i got the same error.

Regards,

**Comment by lutskevich** [comment]
@Jonathan-Caruana just daily cron build (they make it every 12 hours), without fixes

**Comment by Jonathan-Caruana** [comment]
@lutskevich oh ok ! Thank you for that information

**Comment by lutskevich** [comment]
@aporwal3 did your awx deployed succesfull?
Image with downgraded version of cryptography need to apply as Control Plane Execution Environment (used for project update)
![image](https://user-images.githubusercontent.com/1433297/165524497-df9566d8-4af4-4f62-832e-5943252882db.png)

**Comment by lutskevich** [comment]
https://github.com/ansible/awx-operator#default-execution-environments-from-private-registries - hope it helps

Problem is incompatibility of openssl and cryptography versions - https://github.com/pyca/cryptography/issues/7126

**Comment by JustInVTime** [comment]
For people who don't want to/ can't use a private registry, I've created a (temporary) public image on docker hub.
https://hub.docker.com/r/deployablenl/awx-controlplane-ee

Just add the following to your spec section of you deployment:

```
apiVersion: "awx.ansible.com/v1beta1"
kind: AWX
spec:
  ....
  control_plane_ee_image: deployablenl/awx-controlplane-ee:latest
  ....
```

**Comment by seadoox** [comment]
Seeing the same issue and error messages when syncing projects that are stored in gitlab that have a galaxy requirements file. The controlplane-ee gets the error message when it tries to run the ansible-galaxy collection install command.

**Comment by coreydeli** [comment]
Same issue here, I was able to get around it a little bit with @JustInVTime and his temp control plane but this never should have released like this. We should have versions to drop back to, the most recent version we can select outside of "latest" is 0.6.0

**Comment by aporwal3** [comment]
@lutskevich @JustInVTime I have deleted AWX object and recreated it with below specification
 
```
apiVersion: "awx.ansible.com/v1beta1"
kind: AWX
spec:
 ....
 control_plane_ee_image: deployablenl/awx-controlplane-ee:latest
```


But still control plane image is getting picked as 
quay.io/ansible/awx-ee:latest
Am I missing anything ? Do I need to do any other changes as well ?

**Comment by lutskevich** [comment]
@aporwal3 look at kubectl logs -f deployments/awx-operator-controller-manager -c awx-manager, and check your docker-registry secret, try to place it in image_pull_secret.
```
apiVersion: "awx.ansible.com/v1beta1"
kind: AWX
spec:
 ....
   image_pull_secret: gitreg-secret
   control_plane_ee_image: deployablenl/awx-controlplane-ee:latest
   #ee_pull_credentials_secret: gitreg-ee-secret # doesn't work
```
`kubectl create secret docker-registry gitreg-secret --docker-server='url' --docker-username='username' --docker-password='password/token'`

That's all

**Comment by Jonathan-Caruana** [comment]
> For people who don't want to/ can't use a private registry, I've created a (temporary) public image on docker hub. https://hub.docker.com/r/deployablenl/awx-controlplane-ee
> 
> Just add the following to your spec section of you deployment:
> 
> ```
> apiVersion: "awx.ansible.com/v1beta1"
> kind: AWX
> spec:
>   ....
>   control_plane_ee_image: deployablenl/awx-controlplane-ee:latest
>   ....
> ```

Thanks @JustInVTime it's work like a charm

**Comment by lutskevich** [comment]
Issue about ee_pull_credentials_sectet https://github.com/ansible/awx-operator/issues/818

**Comment by eselvam** [comment]
Yes.  we also got same error: awx AttributeError: module 'lib' has no attribute 'X509_V_FLAG_CB_ISSUER_CHECK

as we used custom Ee to over come this.  shall we know when we can expect AWX EE to be patched to fix this?

**Comment by aporwal3** [comment]
@lutskevich @JustInVTime Now in my case, customed ee image is getting succesfully pulled and getting assinged to awx-ee container, as shown below
![image](https://user-images.githubusercontent.com/13012295/165711010-90ebf34e-1510-4e23-9419-0c87a7747dfa.png)


But when I open AWX it is not getting reflected on Execution Environments
![image](https://user-images.githubusercontent.com/13012295/165711213-72d0c52b-377f-48a5-a875-33f058fc51a0.png)

Also there are no error on manager logs. 

So, can you please advise on the same, as in what might be going wrong. 

Do I need to delete the AWX operator and create it all together again ?

**Comment by lutskevich** [comment]
@aporwal3 Looks like you're doing right, i don't know what's wrong
> Do I need to delete the AWX operator and create it all together again ?

Nope, just apply manifest

**Comment by aporwal3** [comment]
@lutskevich : Tried even a fresh install but the customized image is now getting reflected on AWX .. do you know when the fix will be promoted to the original awx-ee image ? Or do you have anything in mind, which can help me to get unblock on this issue ?

**Comment by nicolaibaralmueller** [comment]
Same issue here when upgrading from 19.4.0 to 21.0.0

**Comment by fculpo** [comment]
same

**Comment by aporwal3** [comment]
Mine issue got resolved when I updated awx operator to 0.21.0, earlier I was working on 0.20.2

**Comment by DaDenniX** [comment]
We are also affected by this. Using AWX 20.1.0, AWX EE is 0.6.0 but control plane is awx-ee:latest. Project sync is not possible (with collection installation)

**Comment by chuegel** [comment]
Adding `image_pull_policy: Always`to the awx spec solved the issue for us:

```bash
apiVersion: awx.ansible.com/v1beta1
kind: AWX
metadata:
   name: awx-demo
spec:
   image_pull_policy: Always
```

**Comment by iMikeG6** [comment]
Seriously AWX Team, please test your product before calling it "release".

I confirm the bug as well on our side, same as @JonathanG3 
`  This can be caused if the version of ansible-runner in your execution environment is out of date.`

awx-ee:0.6.0 is broken on release 21.0

Thank you @JustInVTime for providing this fix. However, it'll be nice to have this nice and easy fix in the official image.

**Comment by chuegel** [comment]
While `image_pull_policy: Always`fixes the awx-ee, the error still persists when building custom execution environments:

```bash
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/ansible_runner/__main__.py", line 901, in main
    res = run(**run_options)
  File "/usr/local/lib/python3.8/site-packages/ansible_runner/interface.py", line 216, in run
    r.run()
  File "/usr/local/lib/python3.8/site-packages/ansible_runner/streaming.py", line 130, in run
    r = ansible_runner.interface.run(**self.kwargs)
  File "/usr/local/lib/python3.8/site-packages/ansible_runner/interface.py", line 215, in run
    r = init_runner(**kwargs)
  File "/usr/local/lib/python3.8/site-packages/ansible_runner/interface.py", line 123, in init_runner
    rc = RunnerConfig(**kwargs)
  File "/usr/local/lib/python3.8/site-packages/ansible_runner/config/runner.py", line 79, in __init__
    super(RunnerConfig, self).__init__(private_data_dir, **kwargs)
TypeError: __init__() got an unexpected keyword argument 'suppress_env_files'


  This can be caused if the version of ansible-runner in your execution environment is out of date.
```
execution-environment.yml:
```yaml
---
version: 1
dependencies:
  galaxy: requirements.yml
  python: requirements.txt

additional_build_steps:
  prepend: |
    RUN pip3 install --upgrade pip setuptools
  append:
    - RUN ls -la /etc
```
requirements.txt:
```yaml
junos-eznc
jxmlease
xmltodict
jsnapy
psutil
jmespath
icdiff==1.9.1
cryptography>=35.0,<37.0.0
pyopenssl
ansible-runner
```
Any hints are appreciated.

** Edit **
OK its fixed now. It turned out that our build environment didn't had the latest awx-ee image . Nasty *latest* tag :/

**Comment by lutskevich** [comment]
quay.io/ansible/awx-ee:latest - fixed

**Comment by damianszczepanik** [comment]
I have local repository so I have downloaded latest version, put tag ``latest`` to that version and finally modified ``kustomization.yaml``:

```
images:
  - name: quay.io/ansible/awx-ee
      newName: registry.local/external/ansible/awx-ee
      newTag: latest
```

**Comment by deeco** [comment]
why are these issues closed when the issue remains and people using workarounds from the official readme guides ?

**Comment by coreydeli** [comment]
Technically the issue came back after being fixed. I would look into building your own honestly, I did it and it was very easy to do and solved this issue for us long term.
